lib/spreedly/environment.rb

```diff
@@ -174,14 +174,27 @@ def auth_purchase_body(amount, payment_method_token, options)
       build_xml_request('transaction') do |doc|
         doc.amount amount
         doc.currency_code(options[:currency_code] || currency_code)
-        doc.payment_method_token(payment_method_token)
+        add_payment_token(doc, payment_method_token, options)
         add_to_doc(doc, options, :retain_on_success)
         add_to_doc(doc, options, :stored_credential_initiator)
         add_to_doc(doc, options, :stored_credential_reason_type)
         add_extra_options_for_basic_ops(doc, options)
       end
     end
 
+    def add_payment_token(doc, payment_method_token, options = {})
+      if options[:payment_method] == :google_pay
+        doc << <<~XML
+          <google_pay>
+            <payment_data><![CDATA[#{payment_method_token}]]></payment_data>
+            #{'<test_card_number>4111111111111111</test_card_number>' if options[:test_mode]}
```

Comment:
@guillec Is `options[:test_mode]` always false for production, and true for development, beta, test?

Reply:
Yes Tom, Here is how we are setting. 
`def payment_profile_specific_fields
    options = {}
    return options unless payment_profile.digital_wallet?

    options[:payment_method] = :google_pay if payment_profile.google_pay?
    options[:payment_method] = :apple_pay if payment_profile.apple_pay?
    options[:test_mode] = !Rails.env.production? # This is used for adding test card nubmer for Digital wallets in gem
    options
  end`